credoai/modules/credoai_metrics.py

```diff
@@ -227,3 +227,52 @@ def ks_statistic(y_true, y_pred) -> float:
     ks_stat = st.ks_2samp(y_true, y_pred).statistic
 
     return ks_stat
+
+
+def interpolate_increasing_thresholds(series1, series2, lib_thresholds):
+    series1_out = list()
+    series2_out = list()
+    interpolated_thresholds = np.arange(0, 1, min(0.001, 1 / len(lib_thresholds)))
+
+    for t in interpolated_thresholds:
+        if t >= lib_thresholds[0]:
+            lib_thresholds.pop(0)
+            series1.pop(0)
+            series2.pop(0)
+        series1_out.append(series1[0])
+        series2_out.append(series2[0])
+
+    return series1_out, series2_out, interpolated_thresholds
+
+
+def interpolate_decreasing_thresholds(series1, series2, lib_thresholds):
+    series1_out = list()
+    series2_out = list()
+    interpolated_thresholds = np.arange(
+        max(lib_thresholds), 0, -1 * min(0.001, 1 / len(lib_thresholds))
```

Comment:
Yes & no. I ran into this issue yesterday. Consider snippet below.

`lib_thresholds == 2` since `max(y_score) + 1 = 1 + 1` and `interpolated_thresholds = [1, 1 - quant, ...]`.

If we set the `np.arange()` to start at 1 and count down, then the `if t <= lib_thresholds[0]` line will always be false (the front is = 2).

We could pop the front by default, but there could be some other metric we use in the future that has different behavior and it's not that much overhead to set the `arange` arguments according to the inputs.

```
def interpolate_decreasing_thresholds2(lib_thresholds, *series, quantization=0.001):
    out = [list() for i in series]
    interpolated_thresholds = np.arange(1, 0 - quantization, quantization)

    for t in interpolated_thresholds:
        for i, s in enumerate(out):
            s.append(series[i][0])
        if t <= lib_thresholds[0]:
            lib_thresholds.pop(0)
            for s in series:
                s.pop(0)

    return out + [interpolated_thresholds]
```

Reply:
I was thinking that you are devoting half of your data table to an arbitrary upper threshold. I'm not really following your point about `t <= lib_thresholds[0]` always being false. Wouldn't it be true from the beginning? 1<=2

Reply:
Sorry, I was meant to say True --> you end up getting an "accessing an empty array" error if we're not careful. Basically we pop too much stuff out of the relevant lists.

Here's a cooked up example:
<img width="1025" alt="Screen Shot 2022-10-04 at 3 10 03 PM" src="https://user-images.githubusercontent.com/114082449/193905300-b0822ed2-cecd-494b-bbc5-527f240ba6ce.png">

Reply:
It might be the case that we can trust one bound but not the other (e.g. upper bound for decreasing list and lower bound for increasing list) but it seems brittle to hard-code it.